docs/writing-rules/data-flow/taint-mode.md

```diff
@@ -126,18 +152,300 @@ if cond:
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=yyjrx" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+In essence, using a control source we can define a context, from which Semgrep will detect if a call to some other code (i.e., the sink) can be reached.
+
+:::note
+Using [taint labels](#taint-labels-pro-) it is possible to combine both data and control sources in the same rule.
+:::
+
+## Sanitizers
+
+A taint sanitizer is specified by a pattern. Like in a search-mode rule, you can start this pattern with one of the following keys: `pattern`, `patterns`, `pattern-either`, `pattern-regex`. Note that **any** subexpression that is matched by this pattern will be regarded as sanitized.
+
+In addition, taint sanitizers accept the following options:
+
+| Option            | Type                      | Default | Description                                                            |
+| :-----------------|:------------------------- | :------ | :--------------------------------------------------------------------- |
+| `exact`           | {`false`, `true`}         | `false` | See [_Exact sanitizers_](#exact-sanitizers).                              |
+| `by-side-effect`  | {`false`, `true`, `only`} | `false` | See [_Sanitizers by side-effect_](#sanitizers-by-side-effect).         |
+
+Example:
+
+```yaml
+pattern-sanitizers:
+- pattern: sanitize(...)
+```
+
+### Exact sanitizers
+
+Given the sanitizer specification below, and a piece of code such as `sanitize(sink("taint"))`, the call `sink("taint")` is **not** reported.
+
+```yaml
+pattern-sanitizers:
+- pattern: sanitize(...)
+```
+
+The reason is that the pattern `sanitize(...)` matches all of `sanitize(sink("taint"))`, and that makes Semgrep consider every subexpression in that piece of code as being sanitized. In particular, `"taint"` is considered to be sanitized!
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=v83Rb" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+This is the default for historical reasons, but it may change in the future.
+
+It is possible to instruct Semgrep to only consider as sanitized the "exact" matches of a sanitizer pattern by setting `exact: true`:
+
+```yaml
+pattern-sanitizers:
+- pattern: sanitize(...)
+  exact: true
+```
+
+Once the source is "exact", Semgrep will no longer consider subexpressions as sanitized, and `sink("taint")` inside `sanitize(sink("taint"))` will be reported as a tainted sink.
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=Zqz8o" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+For many rules this distinction is not very meaningful because it does not always make sense that a sink occurs inside the arguments of a sanitizer function.
+
+:::note
+If one of your rules relies on non-exact matching of sanitizers, we advice you to make it explicit with `exact: false`, even if it is the current default, so that your rule does not break if the default changes.
```

Comment:
```suggestion
If one of your rules relies on non-exact matching of sanitizers, We at Semgrep advise you to make it explicit with `exact: false`, even if it is the current default, so that your rule does not break if the default changes.
```
https://developers.google.com/style/word-list#we

Just making it clear who "we" are, at least at this point in the doc.